File: backends/ruby-rails/app/domain/gate/value_objects.rb
module Domain
  module Gate
    IssueVo  = Struct.new(:token, keyword_init: true)
    VerifyVo = Struct.new(:claims, keyword_init: true)

    module CacheRepository
      def get_jwt(identifier, member_id)              = raise NotImplementedError
      def put_jwt(identifier, member_id, token, ttl)  = raise NotImplementedError
    end
  end
end

File: backends/go-beego/internal/domain/gate/value_objects.go
package gate

type IssueVo struct {
	Token string
}

type VerifyVo struct {
	Claims map[string]interface{}
}

type CacheRepository interface {
	GetJwt(identifier, memberID string) (string, error)
	PutJwt(identifier, memberID, token string, ttl int) error
}
